scst/src/scst_sysfs.c

```diff
@@ -7593,6 +7593,80 @@ static struct kobj_attribute scst_last_sysfs_mgmt_res_attr =
 	__ATTR(last_sysfs_mgmt_res, S_IRUGO,
 		scst_last_sysfs_mgmt_res_show, NULL);
 
+static ssize_t scst_cluster_name_show(struct kobject *kobj,
+	struct kobj_attribute *attr, char *buf)
+{
+	int res;
+
+	TRACE_ENTRY();
+
+	if (scst_dlm_cluster_name != NULL)
+		res = sprintf(buf, "%s\n%s", scst_dlm_cluster_name,
+			SCST_SYSFS_KEY_MARK "\n");
+	else
+		res = 0;
+
+	TRACE_EXIT_RES(res);
+	return res;
+}
+
+static ssize_t scst_cluster_name_store(struct kobject *kobj,
+	struct kobj_attribute *attr, const char *buf, size_t count)
+{
+	int res;
+	char *p;
+	int len;
+
+	TRACE_ENTRY();
+
+	if ((buf == NULL) || (count == 0)) {
+		res = 0;
+		goto out;
+	}
+
+	len = strnlen(buf, count);
+	if (buf[count-1] == '\n')
+		len--;
+
+	if (len == 0) {
+		kfree(scst_dlm_cluster_name);
+		scst_dlm_cluster_name = NULL;
+		goto out_done;
+	}
+
+	if (len >= DLM_LOCKSPACE_LEN) {
+		PRINT_ERROR("cluster_name string too long (len %d)", len);
+		res = -EINVAL;
+		goto out;
+	}
+
+	p = kmalloc(len+1, GFP_KERNEL);
```

Comment:
This is an opportunity for `kstrndup`

Reply:
Thanks for the feedback.  I've update the PR to incorporate your comments.